### PR TITLE
 Filter Icon of Table

### LINKED ISSUE #5287: Custom filter, the input box has a value, the icon does not change color
Custom filter, the input box has a value, the icon does not change color,does not satisfy the user experience.

### PR DESCRIPTION
**Pull request by RaoHai**
+ close #5287
+ add `isFiltered` props for mark dataSource is filterd
+ improve CustomFilter Demo

### CODE REVIEW

**Comment by afc163** [comment]
1. should be put inside column object.
2. `isFiltered` => `filtered`, match with `filteredValue`.
3. PR to `feature-2.9`, it is a new feature.

```jsx
<Table columns={[{ filtered: true }]} />
```

**Comment by RaoHai** [comment]
done

File: components/table/demo/custom-filter-panel.md [hunk 1]
```diff
@@ -43,6 +43,7 @@ class App extends React.Component {
     filterDropdownVisible: false,
     data,
     searchText: '',
+    isFiltered: false,
```
Comment (afc163) [inline components/table/demo/custom-filter-panel.md]: filtered

File: components/table/index.zh-CN.md [hunk 2]
```diff
@@ -94,6 +94,7 @@ const columns = [{
 | filterDropdownVisible | 用于控制自定义筛选菜单是否可见 | boolean | - |
 | onFilterDropdownVisibleChange | 自定义筛选菜单可见变化时调用 | function(visible) {} | - |
 | filteredValue | 筛选的受控属性，外界可用此控制列的筛选状态，值为已筛选的 value 数组 | string[] | - |
+| filtered | 标识数据是否经过过滤 | boolean | false |
```
Comment (afc163) [inline components/table/index.zh-CN.md]: 加一句『筛选图标会高亮』

File: components/table/filterDropdown.tsx [hunk 3]
```diff
@@ -209,7 +210,7 @@ export default class FilterMenu extends React.Component<FilterMenuProps, any> {
       </FilterDropdownMenuWrapper>
     );
 
-    const dropdownSelectedClass = (this.props.selectedKeys.length > 0)
+    const dropdownSelectedClass = (this.props.selectedKeys.length > 0 || this.props.filtered)
```
Comment (afc163) [inline components/table/filterDropdown.tsx]: 如果有 `filtered`，则只看 `filtered` 吧。

**Comment by afc163** [comment]
想了下，不如把 `filtered` 改为 `filterIcon`，用户自行负责这个图标的渲染即可，也同时满足自定义图标的需求。

**Comment by afc163** [comment]
rebase

**Pull request merged**